parse/lex.go

```diff
@@ -249,23 +251,46 @@ func defaultStateFn(s *state, e emitter) stateFn {
 	return defaultStateFn
 }
 
-// TODO: implement me w/ test cases :-)
+// TODO: handle error case. panic isn't enough
+//
 // NumberStateFn scans an alphanumeric. ex) 123, 4001, 232
 // After reading Number, it returns DefaultStateFn.
+// number = { decimal_digit }
 func numberStateFn(s *state, e emitter) stateFn {
+	if r := s.peek(); !(r == '+' || r == '-' || ('0' <= r && r <= '9')) {
+		panic("Invalid Function call : numberStateFn")
+	}
+	//accept first character
+	s.next()
+
+	digits := "0123456789"
+	for strings.ContainsRune(digits, s.peek()) {
+		s.next()
+	}
 
+	e.emit(s.cut(Int))
 	return defaultStateFn
```

Comment:
Is there no change in defaultStateFn?

Reply:
yes. can say only state might change ( e.g., start, end pos)